mindmeld/components/entity_resolver.py

```diff
@@ -562,38 +776,829 @@ def _predict_exact_match(self, entity):
+                    output_type
                 )
-                if not self._es_client.indices.exists(index=scoped_index_name):
-                    self.fit()
+            except OSError:
+                logger.error(
+                    "Could not initialize the model name through sentence-transformers models in "
+                    "huggingface; Checking `%s` model directly in huggingface models",
+                    self._model_pretrained_name_or_abspath)
+                try:
+                    transformer_model, pooling_model, sbert_model = \
+                        self._get_transformer_and_pooler(
+                            self._model_pretrained_name_or_abspath,
+                            output_type
+                        )
+                except OSError as e:
+                    logger.error("Could not initialize the model name through huggingface models. "
+                                 "Please check the model name and retry.")
+                    raise ValueError from e
+            if self.quantize_model:
+                if not _is_module_available("torch"):
+                    raise ImportError("`torch` library required to quantize models") from None
+
+                torch_qint8 = _get_module_or_attr("torch", "qint8")
+                torch_nn_linear = _get_module_or_attr("torch.nn", "Linear")
+
+                transformer_model = _get_module_or_attr("torch.quantization", "quantize_dynamic")(
+                    transformer_model, {torch_nn_linear}, dtype=torch_qint8
+                ) if transformer_model else None
+                pooling_model = _get_module_or_attr("torch.quantization", "quantize_dynamic")(
+                    pooling_model, {torch_nn_linear}, dtype=torch_qint8
+                ) if pooling_model else None
+                sbert_model = _get_module_or_attr("torch.quantization", "quantize_dynamic")(
+                    sbert_model, {torch_nn_linear}, dtype=torch_qint8
+                ) if sbert_model else None
+
+            setattr(SentenceBertCosSimEntityResolver, "TRANSFORMER_MODEL", transformer_model)
+            setattr(SentenceBertCosSimEntityResolver, "POOLING_MODEL", pooling_model)
+            setattr(SentenceBertCosSimEntityResolver, "SBERT_MODEL", sbert_model)
+
+        self.transformer_model = getattr(SentenceBertCosSimEntityResolver, "TRANSFORMER_MODEL")
+        self.pooling_model = getattr(SentenceBertCosSimEntityResolver, "POOLING_MODEL")
+        self.sbert_model = getattr(SentenceBertCosSimEntityResolver, "SBERT_MODEL")
+
+        msg = "bert embeddings are cached for entity_type: `%s` " \
+              "for quicker entity resolution; consumes some disk space"
+        logger.info(msg, self.type)
+
+    @property
+    def default_er_config(self):
+        defaults = {
+            "model_settings": {
+                "pretrained_name_or_abspath": "distilbert-base-nli-stsb-mean-tokens",
```

Comment:
I think it makes sense to make this key into two:
```
name: "distilbert-base-nli-stsb-mean-tokens",
path: "abc/xyz",
```
where the path key is optional

Reply:
this is kept so as to make things similar to huggingface models, which might be more easier for an NLP developer to follow, especially given that we re-use huggingface models in the sbert based resolver. And additionally, we don't need to have two keys because we just pass-in the combined key into huggingface AutoXXX factory and it internally resolves it, else we need to resolve it which adds more boilerplate code!